### PR TITLE
Various small additions

### PR DESCRIPTION
**Pull request by wossnameGitHub**
At first this may seem like a complicated Merge Request, but if you check the commits individually I think you'll quickly realise these are very simple changes.

- changed order of some code blocks in entrance_rando.py
- changed order of some code blocks in utils.py
- the 2 highjack_transitions functions are moved from entrance_rando.py to utils.py
- temp_disabled_exits is added (mainly used in the graph)
- VALLEY_OF_THE_SPIRITS renamed to VALLEY_OF_SPIRITS
- adjusted some of the spacing in the generated graphml file

### CODE REVIEW

File: Dolphin scripts/Entrance Randomizer/lib/utils.py [hunk 1]
```diff
@@ -105,6 +63,83 @@ def follow_pointer_path(ppath: Sequence[int]):
...
+            "highjack_transition |",
+            f"From: {hex(state.current_area_old)},",
+            f"To: {hex(state.current_area_new)}.",
+            f"Redirecting to: {hex(redirect)}",
+        )
+        memory.write_u32(ADDRESSES.current_area, redirect)
+        return True
+    return False
+
+
```
Comment (Avasam) [inline Dolphin scripts/Entrance Randomizer/lib/utils.py:91]: Whilst I agree that `highjack_transition` definitely has a general enough use to go in utils, `highjack_transition_rando` does belong specifically in the entrance rando module.

**Review by Avasam — CHANGES_REQUESTED** [review]
I'll finish this review and give a hand with static typing when I get back home.

But I'm not sure why so much stuff needs to be moved around. Does it even reduce forward references (making it so methods are declared before they're used visually)? It makes the git history messier, harder to find and track why things were added/removed and by whom (by using the "blame" tool. Also easily creates conflicts.

Otherwise the functional changes seem fine. The graph changes are a good improvement.

File: Dolphin scripts/Entrance Randomizer/lib/utils.py (hunk 1 above)
Comment (wossnameGitHub) [reply Dolphin scripts/Entrance Randomizer/lib/utils.py:91]: About moving stuff around, it makes the code look cleaner and more readable if things are written in a logical order, compared to just shoving functions in a file without regard for their actual contents.
Also this moving around neatly ties in the work I've done in the upcoming merges. I thought I would do you a favour by splitting up the moving around (this MR) from the actual rewriting and adding new code (the upcoming MR's).

About the highjack functions, I noticed that 94% of all code in entrance_rando.py was related to creating the transitions_map. Then 5% of the code was those 2 highjack functions, and then there's 1% other stuff.
I thought it would make sense to seperate the creation of transitions_map (which is called once when you start up the rando) from stuff that doesn't have anything to do with creating the transitions_map. Especially if those other things are called constantly during runtime.
That means that the highjack functions had to go. 1) They don't contribute to the creation of transitions_map, and 2) They are constantly called during runtime, which isn't the case for anything else in entrance_rando.py.
As for where to move these 2 functions to, I thought it would make a lot of sense to put them in the same file, as they are very similar functions. So I basically had the choice: 1) make a new file and shove them in there, or 2) put them in one of the existing files. I chose option 2 and put them in utils.py, because utils.py already had a couple of functions in it that were called during runtime (closed door teleporting and the Apu Illapu Shrine + Scorpion Temple situations).

So long story short:
1) I think the new system is very clean, easy to understand and logical
2) I think the old system placed the highjack functions in completely illogical locations
3) I think the new location for the highjack functions makes a lot of sense
4) I moved things around this MR because I have a lot of big changes in the upcoming MR's, but at least this way I won't need to move stuff around in those other MR's

Comment (wossnameGitHub) [reply Dolphin scripts/Entrance Randomizer/lib/utils.py:91]: Also who is this creating conflicts for?
Is anyone else doing big work on entrance_rando.py right now besides me? If so I wasn't aware.

Comment (Avasam) [reply Dolphin scripts/Entrance Randomizer/lib/utils.py:91]: > Especially if those other things are called constantly during runtime.
> That means that the highjack functions had to go. 1) They don't contribute to the creation of transitions_map, and 2) They are constantly called during runtime, which isn't the case for anything else in entrance_rando.py.

It really doesn't matter whether something is called once or multiple times in a module. It's more about separation of concerns, domains, and avoiding cyclical imports.
"utils" by nature is a vague naming (as any helpful method or structure is a util) so I'll often prefer moving something *out* of utils rather than *in* if it can be grouped logically into its own module.
`highjack_transition_rando` is very specific to entrance randomization

> so I basically had the choice: 1) make a new file and shove them in there, or 2) put them in one of the existing files. 

If anything, moving stuff related to transition/entrance handling out of the vaguely named "utils" would be preferable. As long as it's not specific to, let's say, entrance rando, item rando, shaman shop rando, etc.

> entrance_rando.py was getting really big and needed to be split up sooner or later

If you think it's getting too big, you can make `entrance_rando` a subpackage (a folder with `__init__.py`). ~500 lines really isn't that much for Python modules due to how easy to fall in cyclical import traps. (compared to let's say JS/TS/C# where I'll go as far as having 1 model/component/interface/class per file and hard-limit at 500 lines)

> Also who is this creating conflicts for?

Yourself 
image

Comment (Avasam) [reply Dolphin scripts/Entrance Randomizer/lib/utils.py:91]: (note, other than, `highjack_transition_rando`, which I really believe doesn't belong in `utils`, I'm not asking you to revert any of the stuff you moved, I'll accept the PR as-is. Just wanted to know your reasoning about it)

File: Dolphin scripts/Entrance Randomizer/lib/entrance_rando.py [hunk 2]
```diff
@@ -28,6 +27,17 @@ class Choice(IntEnum):
     LevelCRC.PENGUIN_TEMPLE,
 )
 
+one_way_exits = (
```
Comment (Avasam) [inline Dolphin scripts/Entrance Randomizer/lib/entrance_rando.py:30]: Whilst we're moving stuff, knowing which exits are one-way isn't specific to the rando, you can move that out of `entrance_rando` and with `TEMPLES_WITH_FIGHT`, `SOFTLOCKABLE_ENTRANCES` if you want (don't have to do that in this PR, it was already here anyway)

I think it belongs even better directly encoded in `transition_infos`, but no need to scope creep on this PR.

File: Dolphin scripts/Entrance Randomizer/lib/graph_creation.py [hunk 3]
```diff
@@ -109,32 +109,36 @@ def create_edges(transitions_map: Mapping[tuple[int, int], tuple[int, int]]):
...
 ):
-    graphml_text = f"""\
-<?xml version="1.0" encoding="UTF-8"?>
-<graphml>
-  <graph id="Graph" uidGraph="1" uidEdge="1">
-    {create_vertices(transitions_map, starting_area)}
-    {create_edges(transitions_map)}
-  </graph>
-</graphml>"""
+    all_transitions: Mapping[tuple[int, int], tuple[int, int]] = transitions_map.copy()
```
Comment (Avasam) [inline Dolphin scripts/Entrance Randomizer/lib/graph_creation.py:131]: By using copy, this is locked down to being a dict, you can simply update the expected type

```python
    transitions_map: dict[tuple[int, int], tuple[int, int]],
    temp_disabled_exits: Sequence[tuple[int, int]],
    seed_string: SeedType,
    starting_area: int,
):
    all_transitions = transitions_map.copy()
```

File: Dolphin scripts/Entrance Randomizer/lib/entrance_rando.py (hunk 2 above)
Comment (wossnameGitHub) [reply Dolphin scripts/Entrance Randomizer/lib/entrance_rando.py:30]: I don't want to do that right now.

File: Dolphin scripts/Entrance Randomizer/lib/graph_creation.py (hunk 3 above)
Comment (wossnameGitHub) [reply Dolphin scripts/Entrance Randomizer/lib/graph_creation.py:131]: Upper change applied.

File: Dolphin scripts/Entrance Randomizer/lib/utils.py (hunk 1 above)
Comment (wossnameGitHub) [reply Dolphin scripts/Entrance Randomizer/lib/utils.py:91]: Okay, so how about a compromise?
I'll move `highjack_transition_rando` back to `entrance_rando.py`, but I leave everything else in the location it is now.
Deal?

Comment (Avasam) [reply Dolphin scripts/Entrance Randomizer/lib/utils.py:91]: Yep!

**Review by Avasam — APPROVED** [review]
